lib/logstash/environment.rb

```diff
@@ -114,6 +95,42 @@ def bundler_setup!(options = {})
       ::Bundler.setup
     end
 
+    def runtime_jars_root(dir_name, package)
```

Comment:
After discussing with @purbon, let's go with the current strategy. The amount of code plugins need to load their jars should either be a one liner or it could be detected automatically. However this requires a different strategy, so I've opened an issue to deal with this: https://github.com/elastic/logstash/issues/3005